src/services/ui/src/pages/actions/WithdrawPackage.tsx

```diff
@@ -0,0 +1,229 @@
+import { Navigate, useNavigate, useParams } from "react-router-dom";
+import { Button } from "@/components/Inputs";
+import { ConfirmationModal } from "@/components/Modal/ConfirmationModal";
+import { useEffect, useState } from "react";
+import { Action, ItemResult, withdrawPackageSchema } from "shared-types";
+import { FAQ_TARGET, ROUTES } from "@/routes";
+import { PackageActionForm } from "./PackageActionForm";
+import { ActionFormIntro, PackageInfo } from "./common";
+import { z } from "zod";
+import { SubmitHandler, useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import * as I from "@/components/Inputs";
+import { Link } from "react-router-dom";
+import { Alert, LoadingSpinner } from "@/components";
+import { buildActionUrl } from "@/lib";
+import { useGetUser } from "@/api/useGetUser";
+import { useSubmissionService } from "@/api/submissionService";
+
+const withdrawPackageFormSchema = withdrawPackageSchema(
+  z.array(z.instanceof(File))
+);
+type WithdrawPackageFormSchema = z.infer<typeof withdrawPackageFormSchema>;
+type UploadKey = keyof WithdrawPackageFormSchema["attachments"];
+type AttachmentRecipe = {
+  readonly name: UploadKey;
+  readonly label: string;
+  readonly required: boolean;
+};
+
+const attachments: AttachmentRecipe[] = [
+  {
+    name: "supportingDocumentation",
+    label: "Supporting Documentation",
+    required: false,
+  } as const,
+];
+
+const handler: SubmitHandler<WithdrawPackageFormSchema> = (data) =>
+  console.log(data);
+
+const WithdrawPackageForm: React.FC = ({ item }: { item?: ItemResult }) => {
+  const navigate = useNavigate();
+  const { id, type } = useParams<{ id: string; type: Action }>();
+  const { data: user } = useGetUser();
+  const form = useForm<WithdrawPackageFormSchema>({
+    resolver: zodResolver(withdrawPackageFormSchema),
+  });
+  const { mutate, isLoading, isSuccess, error } = useSubmissionService<{
+    id: string;
+  }>({
+    data: { id: id! },
```

Comment:
Data from the form isn't being passed to the `submit()` function underneath this hook interface, and since this uses a `form` tag and `react-hook-form`, we can still make use of the `handleSubmit` pattern that'll pre-validate our form data before it ever reaches the `submit()` function. 

To see how we can set this up, check out [IssueRai.tsx](https://github.com/Enterprise-CMCS/macpro-mako/blob/87bc5f3f3ba1ba3a290b7a8f64c1f6621c873a5a/src/services/ui/src/pages/actions/IssueRai.tsx#L79C3-L94C5)